packages/core/plugins/core-plugins-server-internal/src/plugins_service.ts

```diff
@@ -448,10 +448,16 @@ export class PluginsService
     uiPluginInternalInfo: Map<PluginName, InternalPluginInfo>
   ) {
     for (const [pluginName, pluginInfo] of uiPluginInternalInfo) {
-      deps.http.registerStaticDir(
+      /**
+       * Serve UI from sha-scoped and not-sha-scoped paths to allow time for plugin code to migrate
+       * Eventually we only want to serve from the sha scoped path
+       */
+      [
+        deps.http.staticAssets.getPluginServerPath(pluginName, '{path*}'),
         `/plugins/${pluginName}/assets/{path*}`,
-        pluginInfo.publicAssetsDir
-      );
+      ].forEach((path) => {
+        deps.http.registerStaticDir(path, pluginInfo.publicAssetsDir);
+      });
```

Comment:
Outside of the scope of this PR, but I wonder if this shouldn't be moved to the `core/apps` package alongside all the other plugin/packages route definitions (given we already have the list of plugins there anyway)

Reply:
Yeah, I think that would be nice. Would you mind opening an issue for us to address it?

Reply:
Sure, opened https://github.com/elastic/kibana/issues/176201